lib/util.js

```diff
@@ -19,22 +20,42 @@ const requestDefaults = {
         'User-Agent': USER_AGENT,
     },
 };
-let pooledRequest = request.defaults(requestDefaults);
+
+// Exponential wait: 800ms, 1600ms, 3200ms
+const MAX_RETRIES = 3;
+const RETRY_START_MS = 800;
 
 // Wrap request results in a promise
-function rp (options) {
-    return new Promise((resolve, reject) => {
-        pooledRequest(options, (err, res, body) => {
-            if (err) return reject(err);
-            resolve(res);
-        });
+let rp = promisify(request.defaults(requestDefaults));
+
+function wait (timeout) {
+    return new Promise((resolve) => {
+        setTimeout(() => {
+            resolve()
+        }, timeout);
     });
 }
 
+
+async function requestRetrying (options) {
+    for (let i = 0; i < MAX_RETRIES; i++) {
+        try {
+            return await rp(options);
```

Comment:
We are on Node.js 6 as our base version which AFAIK doesn't support native await keyword